### ISSUE TITLE
DataTable: Using UNSAFE_componentWillUpdate in strict mode is not recommended and may indicate bugs in your code 

### ISSUE THREAD

**Issue by nigellima** [carbon-design-system/carbon#6361]
## DataTable: Using UNSAFE_componentWillUpdate in strict mode is not recommended and may indicate bugs in your code 

--> Hello. I'm using Carbon React components. The DataTable component is triggering a warning, which I'm not sure why (maybe something related to my react version). The component is rendered only when the data is all loaded.

<img width="1354" alt="image" src="https://user-images.githubusercontent.com/14295280/86047212-f2019380-ba24-11ea-8b0e-29a66f01f01e.png">

Here is how I organized it:

This is my `index.tsx`:

```
<DataTable
        showToolbar={false}
        rows={tableRows}
        headers={tableHeaders}
        render={({
          rows, headers, getHeaderProps, getRowProps, getTableProps,
        }): ReactNode => (
          <TableContainer>
            <Table {...getTableProps()} useZebraStyles={false} size="short">
              <TableHead>
                <DetailedTableHeaders
                  getHeaderProps={getHeaderProps}
                  headers={headers}
                  store={store}
                  tableHeadersLen={tableHeaders.length}
                />
              </TableHead>
              <TableBody>
                <DetailedTableRows
                  store={store}
                  colSpan={headers.length + 1}
                  getRowProps={getRowProps}
                  rows={rows}
                  originalRows={tableRows}
                />
              </TableBody>
            </Table>
          </TableContainer>
        )}
      />
```

This is my `DetailedTableHeaders`:

```
<TableRow>
      <TableExpandHeader />
      {headers.map((header, index) => (
        <TableHeader
          ref={headerRefs.current[index]}
          {...getHeaderProps({ header })}
        >
          {header.header}
        </TableHeader>
      ))}
    </TableRow>
```

And here it's my `DetailedTableRows`:

```
return rows.map((row) => {
    const rawCells = row.cells.map((cell, index) => (
      <TableCell
        key={uuid()}
      >
        {cell.value}
      </TableCell>
    ));
    const tableRow = row.id === 'total' ? (
      <TableRow id={row.id} {...getRowProps({ row })} key={uuid()}>
        <TableCell />
        {rawCells}
      </TableRow>
    ) : (
      <TableExpandRow id={row.id} {...getRowProps({ row })} key={uuid()}>
        {rawCells}
      </TableExpandRow>
    );
    return (
      <React.Fragment key={uuid()}>
        {tableRow}
        {row.isExpanded && (
          <DetailedBreakdownExpandedRows
            row={row}
            originalRows={originalRows}
            getRowProps={getRowProps}
          />
        )}
      </React.Fragment>
    );
  });
```

What could be the reason this warning is been triggered?

## What package(s) are you using?

- [ ] `carbon-components`
- [X] `carbon-components-react`
- [X] react: ^16.13.1,

> Please create a reduced test case in CodeSandbox
> I wasnt able to reproduce it in another environment. But the warning says it comes from Carbon's DataTable component

**Comment by nikolasmach23** [comment]
![warning](https://user-images.githubusercontent.com/26468068/86098030-d72b2f80-bab5-11ea-8d74-1455b40d2449.png)

Hi, the same warning. DatePicker too. Using component version "carbon-components-react": "^7.14.0" , "react": "^16.13.1"

**Comment by joshblack** [comment]
Hi there! This is a duplicate of: https://github.com/carbon-design-system/carbon/issues/4902 I believe which was closed over in: https://github.com/carbon-design-system/carbon/pull/6307

Expect to see this in the next release!

**Comment by nigellima** [comment]
> Hi there! This is a duplicate of: #4902 I believe which was closed over in: #6307
> 
> Expect to see this in the next release!

Hi @joshblack 

Thank you very much. Do you know when the next release might be available?

**Comment by joshblack** [comment]
Should be up under latest today!
